src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -123,6 +130,59 @@ static JsonWebKey createJwkFromSettings(Settings settings) throws Exception {
+            jwk.setProperty(EC_X_COORDINATE, Optional.ofNullable(settings.get("ec_x")).orElseThrow(() -> new IllegalArgumentException("Settings for EC x coordinate is missing.")));
+            jwk.setProperty(EC_Y_COORDINATE, Optional.ofNullable(settings.get("ec_y")).orElseThrow(() -> new IllegalArgumentException("Settings for EC y coordinate is missing.")));
+        } else if (AlgorithmUtils.isRsaSign(algorithm)) {
+            jwk.setKeyType(KeyType.RSA);
+            jwk.setAlgorithm(algorithm);
+            jwk.setProperty(RSA_MODULUS, Optional.ofNullable(settings.get("rsa_modulus")).orElseThrow(() -> new IllegalArgumentException("Settings for RSA modulus is missing. Please specify rsa_modulus.")));
+            jwk.setProperty(RSA_PUBLIC_EXP, Optional.ofNullable(settings.get("rsa_public_exp")).orElseThrow(() -> new IllegalArgumentException("Settings for RSA public exponent is missing. Please specify rsa_public_exp.")));
+            jwk.setProperty(RSA_PRIVATE_EXP, Optional.ofNullable(settings.get("rsa_private_exp")).orElseThrow(() -> new IllegalArgumentException("Settings for RSA private exponent is missing. Please specify rsa_private_exp.")));
+
+            //optional settings, we need either none, or all of those
+            List<String> rsaMissingSettings = new ArrayList<>();
+            Optional.ofNullable(settings.get("rsa_first_prime_factor")).ifPresentOrElse(v -> jwk.setProperty(RSA_FIRST_PRIME_FACTOR, v), () -> rsaMissingSettings.add("rsa_first_prime_factor"));
+            Optional.ofNullable(settings.get("rsa_second_prime_factor")).ifPresentOrElse(v -> jwk.setProperty(RSA_SECOND_PRIME_FACTOR, v), () -> rsaMissingSettings.add("rsa_second_prime_factor"));
+            Optional.ofNullable(settings.get("rsa_first_prime_crt")).ifPresentOrElse(v -> jwk.setProperty(RSA_FIRST_PRIME_CRT, v), () -> rsaMissingSettings.add("rsa_first_prime_crt"));
+            Optional.ofNullable(settings.get("rsa_second_prime_crt")).ifPresentOrElse(v -> jwk.setProperty(RSA_SECOND_PRIME_CRT, v), () -> rsaMissingSettings.add("rsa_second_prime_crt"));
+            Optional.ofNullable(settings.get("rsa_first_crt_coefficient")).ifPresentOrElse(v -> jwk.setProperty(RSA_FIRST_CRT_COEFFICIENT, v), () -> rsaMissingSettings.add("rsa_first_crt_coefficient"));
+            if (rsaMissingSettings.size() != 0 && rsaMissingSettings.size() != 5) {
+                //incorrect optional settings
+                throw new IllegalArgumentException("Settings for RSA key is missing, missing properties: " + String.join(", ", rsaMissingSettings));
+            }
+        } else {
+            throw new IllegalArgumentException("Signing algorithm not recognized: " + algorithm);
+        }
+        return jwk;
+    }
+
+    private static String calcCurve(String algorithm) {
+        switch (algorithm) {
+            case ES_SHA_256_ALGO:
+                return EC_CURVE_P256;
+            case ES_SHA_384_ALGO:
```

Comment:
This is the threshold for top secret may be worth documenting as a recommended level when in doubt